meta/meta.go

```diff
@@ -701,15 +703,36 @@ func (m *Meta) SetSchemaDiff(diff *model.SchemaDiff) error {
 	return errors.Trace(err)
 }
 
+// DropPartition drops partition in table.
+func (m *Meta) DropPartition(dbID int64, pid int64, delAutoID bool) error {
+	// Check if db exists.
+	dbKey := m.dbKey(dbID)
+	if err := m.checkDBExists(dbKey); err != nil {
+		return errors.Trace(err)
+	}
+
+	tableKey := m.tableKey(pid)
+	if err := m.txn.HDel(dbKey, tableKey); err != nil {
+		return errors.Trace(err)
+	}
+	if delAutoID {
+		if err := m.txn.HDel(dbKey, m.autoTableIDKey(pid)); err != nil {
```

Comment:
autoTableIDKey belongs to table or partition? @zimulala

Reply:
Done.